### PR TITLE
add the ability to download kind/kubectl on macOS GHA runners

### PR DESCRIPTION
**Pull request by nicks**
Important note - This doesn't setup Docker on MacOS. To setup Docker, you either need to use the Docker Desktop GHA (https://github.com/docker/desktop-action), or you need to connect your MacOS runner to a remote Docker Daemon.

### CODE REVIEW

File: kind.sh [hunk 1]
```diff
@@ -56,14 +56,27 @@ main() {
...
+    case $(uname) in
+        Darwin) os="darwin" ;;
     esac
-    local cache_dir="${RUNNER_TOOL_CACHE}/kind/${version}/${arch}"
+
+    local arch
+    if [[ "$os" == "darwin" ]]; then
+        case $(uname -m) in
+            x86_64)         arch="amd64" ;;
+            arm)            arch="arm64" ;;
```
Comment (musse) [inline kind.sh:68]: On my Macbook with Apple Silicon, `uname -m` returns `arm64`, not `arm`.

Comment (nicks) [reply kind.sh:68]: ahhhh thanks, i changed it to arm|arm64 to be safe

**Review by cpanato — CHANGES_REQUESTED** [review]
lgtm

can you rebase and maybe add a test case that runs on macos?

thanks for this PR and very sorry for the delay

**Comment by nicks** [comment]
added a macos test rule

**Comment by cpanato** [comment]
@nicks the new mac testing is failing

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -68,6 +68,21 @@ jobs:
       - name: Test kind works and there is no cluster started
         run: |
           [[ $(kind get clusters | wc -l) -eq 0 ]]
+          
+  test-with-macos-install-only:
+    runs-on: macos-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
```
Comment (musse) [inline .github/workflows/test.yaml:76]:
```suggestion

  test-with-macos-install-only:
    runs-on: macos-latest
    steps:
      - name: Checkout
        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1

      - name: Setup docker
        run: |
          brew install docker
          colima start
```

Comment (musse) [reply .github/workflows/test.yaml:76]: Tested this fix on my fork, it works: https://github.com/musse/kind-action/pull/1

Check this thread for details: https://github.com/actions/runner-images/issues/17#issuecomment-1445517870

**Comment by nicks** [comment]
the test failure might be revealing an existing bug in the cleanup script?

currently it always runs `kind delete`, even if the action has `install_only` set to true.

arguably, if `install_only` is set to true, then it's not responsible for creating a cluster, so also shouldn't be responsible for cleaning one up?

for now, i set `ignore_failed_clean` on the new test, which is arguably the "safest" change.

File: .github/workflows/test.yaml (hunk 2 above)
Comment (nicks) [reply .github/workflows/test.yaml:76]: i don't think this is right; you shouldn't need `colima` running if the action is in `install_only` mode

**Pull request closed without merging**